### PR TITLE
[PR] Migrate to `v3`

### LINKED ISSUE #15: Nextra V3
@LuchoTurtle when you have a moment in the next couple of days, please update this project to Nextra V3. 🙏 
https://the-guild.dev/blog/nextra-3

### PR DESCRIPTION
**Pull request by LuchoTurtle**
closes #15 

This PR migrates this package to `v3`. It:
- removes `contentlayer2` and switches to `remark`. Links are now validated with `remark-validate-links`
- updates the `theme`
- updates the `src` folder according to the migration guide at https://the-guild.dev/blog/nextra-3.
- updates dependencies
- removes unnecessary/unused files.
- updated the `README` and `tutorial.md` to add instances of the difference between versions.

### CODE REVIEW

**Comment by LuchoTurtle** [comment]
@nelsonic this should be mergeable. The bottleneck was `contentlayer2`, which was replaced with our linting strategy in the `A` docs repo. I've documented both strategies in the `tutorial.md` file.

File: package.json [hunk 1]
```diff
@@ -32,21 +31,32 @@
     "@types/jest": "^29.5.12",
     "@types/node": "20.14.5",
     "autoprefixer": "^10.4.19",
-    "contentlayer2": "^0.5.0",
+    "babel-jest": "^29.7.0",
     "dotenv": "^16.4.5",
     "fast-glob": "^3.3.2",
+    "glob": "^11.0.0",
```
Comment (nelsonic) [inline package.json:37]: curious why we need both `glob` and `fast-glob` ... 💭

File: postcss.config.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ module.exports = {
     'tailwindcss/nesting': {},
     tailwindcss: {},
     'postcss-lightningcss': {
-      browsers: '>= .25%'
+      browsers: '>= .25% and not dead'
```
Comment (nelsonic) [inline postcss.config.js:8]: curious about this setting. how did you discover it? 💭

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -10,22 +10,22 @@ importers:
     dependencies:
       next:
         specifier: ^14.2.4
-        version: 14.2.4(@babel/core@7.24.7)(@opentelemetry/api@1.9.0)(@playwright/test@1.44.1)(react-dom@18.3.1(react@18.3.1))(react@18.3.1)
+        version: 14.2.15(@babel/core@7.25.8)(@opentelemetry/api@1.9.0)(@playwright/test@1.48.0)(react-dom@18.3.1(react@18.3.1))(react@18.3.1)
```
Comment (nelsonic) [inline pnpm-lock.yaml:13]: Who has the attention span to review a 8.6k changes file? 🤷‍♂️ 
Feel like this is now in the hands of bots to keep secure. 🤖 💭

**Review by nelsonic — APPROVED** [review]
@LuchoTurtle looks good. 👌

**Comment by nelsonic** [comment]
Especially like that this PR has a net _removal_ of lines from the project. 👌

File: postcss.config.js (hunk 2 above)
Comment (LuchoTurtle) [reply postcss.config.js:8]: It was from `nextra`'s source code -> https://github.com/shuding/nextra/blob/4eac7fbbaa30be4ac724c0d0a7a388d43636f534/packages/nextra-theme-docs/postcss.config.js#L8.

File: package.json (hunk 1 above)
Comment (LuchoTurtle) [reply package.json:37]: That's what `Nextra` used. I've changed this in https://github.com/dwyl/nextra-demo/pull/17.

File: postcss.config.js (hunk 2 above)
Comment (nelsonic) [reply postcss.config.js:8]: Nice. thanks for link. 🔗

File: package.json (hunk 1 above)
Comment (nelsonic) [reply package.json:37]: Thanks reviewing now. 👌

**Pull request merged**